src/charm.py

```diff
@@ -69,29 +71,65 @@ def _on_config_changed(self, event) -> None:
         self.model.unit.status = MaintenanceStatus(msg)
         self.configure_workload_container(event)
 
+    def _get_ip_type(self, address) -> str:
+        """Validate a string type: ipv4, ipv6 or invalid as IP address."""
+        try:
+            ip = ipaddress.ip_address(address)
+
+            if isinstance(ip, ipaddress.IPv4Address):
+                return "ipv4"
+            elif isinstance(ip, ipaddress.IPv6Address):
+                return "ipv6"
+        except ValueError:
+            return
+
+    def readlines_reverse(self, qfile):
```

Comment:
Is a docstring and are type hints useful here? Also, does this function need to be in the class or could it be a separate module?